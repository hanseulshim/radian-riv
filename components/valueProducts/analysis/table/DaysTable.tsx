import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'

interface Props {
  tableData: any[]
}

export default function DaysTable({ tableData }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const width = 50
    setColumns([
      {
        Header: 'Days',
        accessor: 'days',
        align: 'right'
      },
      {
        Header: '0-90',
        accessor: '0-90',
        width,
        align: 'right'
      },
      {
        Header: '91-180',
        accessor: '91-180',
        width,
        align: 'right'
      },
      {
        Header: '180-270',
        accessor: '180-270',
        width,
        align: 'right'
      },
      {
        Header: '271-365',
        accessor: '271-365',
        width,
        align: 'right'
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <div className="table-spacer">
      <Table columns={columns} data={data} fetchData={fetchData} width={700} />
    </div>
  )
}
