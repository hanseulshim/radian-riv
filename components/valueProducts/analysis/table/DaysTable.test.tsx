import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import DaysTable from './DaysTable'

describe('DaysTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: []
    }
    component = shallow(<DaysTable {...props} />)
  })
  it('get table spacer container', () => {
    expect(component.find('.table-spacer')).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
