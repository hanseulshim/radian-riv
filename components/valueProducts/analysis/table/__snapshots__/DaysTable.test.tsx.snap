// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysTable should match snapshot 1`] = `
<div
  className="table-spacer"
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
    width={700}
  />
</div>
`;
